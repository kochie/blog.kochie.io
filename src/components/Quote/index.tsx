import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faGlobe,
  faHyphen,
  faQuoteLeft,
  faQuoteRight,
} from '@fortawesome/pro-regular-svg-icons'
import { faTwitter } from '@fortawesome/free-brands-svg-icons'
import React, { PropsWithChildren } from 'react'
import Image from 'next/image'
import Link from 'next/link'

export interface QuoteProps {
  author?: string
  position?: string
  twitter?: string
  src?: string
  web?: string
}

export default function Quote({
  author,
  position,
  twitter,
  children,
  src,
  web,
}: PropsWithChildren<QuoteProps>) {
  return (
    <div className="p-5 my-10 font-light text-2xl">
      <FontAwesomeIcon icon={faQuoteLeft} className="" />
      <span className="italic">{children}</span>
      <div className="flex items-center">
        <FontAwesomeIcon icon={faQuoteRight} className="" />
        {(src || author || position) && (
          <FontAwesomeIcon icon={faHyphen} className="mx-4" />
        )}
        <div className="flex items-center ml-2">
          {src && (
            <Image
              src={src}
              alt={`Photo of ${author}`}
              height={64}
              width={64}
              className="h-16 w-16 rounded-full"
              style={{
                maxWidth: '100%',
                height: 'auto',
              }}
            />
          )}
          <div className="text-base ml-3">
            <span>{author}</span>
            {twitter ? (
              <Link href={twitter} className="ml-2">
                <FontAwesomeIcon
                  icon={faTwitter}
                  className="hover:text-blue-500 transform-gpu duration-300"
                />
              </Link>
            ) : null}
            {web ? (
              <Link href={web} className="ml-2">
                <FontAwesomeIcon
                  icon={faGlobe}
                  className="hover:text-red-500 transform-gpu duration-300"
                />
              </Link>
            ) : null}
            <br />
            <span>{position}</span>
          </div>
        </div>
      </div>
    </div>
  )
}
