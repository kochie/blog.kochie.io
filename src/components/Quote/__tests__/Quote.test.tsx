import { library } from '@fortawesome/fontawesome-svg-core'
import { faQuoteLeft, faQuoteRight } from '@fortawesome/free-solid-svg-icons'
import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import Quote from '..'

describe('Quote Component', () => {
  beforeAll(() => {
    library.add(faQuoteRight, faQuoteLeft)
  })

  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(
        <Quote
          author="test"
          position="test-position"
          src="https://pbs.twimg.com/profile_images/1561629357692465152/7PCEt4on_400x400.jpg"
        >
          A quote
        </Quote>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
