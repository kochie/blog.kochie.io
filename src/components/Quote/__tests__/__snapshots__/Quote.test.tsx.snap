// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component renders correctly 1`] = `
<div
  className="p-5 my-10 font-light text-2xl"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-quote-left "
    data-icon="quote-left"
    data-prefix="far"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 216C0 149.7 53.7 96 120 96h16c13.3 0 24 10.7 24 24s-10.7 24-24 24H120c-39.8 0-72 32.2-72 72v10c5.1-1.3 10.5-2 16-2h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V320 288 216zm48 72v32 32c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16V288c0-8.8-7.2-16-16-16H64c-8.8 0-16 7.2-16 16zm336-16H320c-8.8 0-16 7.2-16 16v32 32c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16V288c0-8.8-7.2-16-16-16zM256 320V288 216c0-66.3 53.7-120 120-120h16c13.3 0 24 10.7 24 24s-10.7 24-24 24H376c-39.8 0-72 32.2-72 72v10c5.1-1.3 10.5-2 16-2h64c35.3 0 64 28.7 64 64v64c0 35.3-28.7 64-64 64H320c-35.3 0-64-28.7-64-64V320z"
      fill="currentColor"
      style={{}}
    />
  </svg>
  <span
    className="italic"
  >
    A quote
  </span>
  <div
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-quote-right "
      data-icon="quote-right"
      data-prefix="far"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M448 296c0 66.3-53.7 120-120 120H312c-13.3 0-24-10.7-24-24s10.7-24 24-24h16c39.8 0 72-32.2 72-72V286c-5.1 1.3-10.5 2-16 2H320c-35.3 0-64-28.7-64-64V160c0-35.3 28.7-64 64-64h64c35.3 0 64 28.7 64 64v32 32 72zm-48-72V192 160c0-8.8-7.2-16-16-16H320c-8.8 0-16 7.2-16 16v64c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16zM64 240h64c8.8 0 16-7.2 16-16V192 160c0-8.8-7.2-16-16-16H64c-8.8 0-16 7.2-16 16v64c0 8.8 7.2 16 16 16zm128-48v32 72c0 66.3-53.7 120-120 120H56c-13.3 0-24-10.7-24-24s10.7-24 24-24H72c39.8 0 72-32.2 72-72V286c-5.1 1.3-10.5 2-16 2H64c-35.3 0-64-28.7-64-64V160c0-35.3 28.7-64 64-64h64c35.3 0 64 28.7 64 64v32z"
        fill="currentColor"
        style={{}}
      />
    </svg>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-hyphen mx-4"
      data-icon="hyphen"
      data-prefix="far"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256c0-13.3 10.7-24 24-24H296c13.3 0 24 10.7 24 24s-10.7 24-24 24H24c-13.3 0-24-10.7-24-24z"
        fill="currentColor"
        style={{}}
      />
    </svg>
    <div
      className="flex items-center ml-2"
    >
      <img
        alt="Photo of test"
        className="h-16 w-16 rounded-full"
        data-nimg="1"
        decoding="async"
        height={64}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1561629357692465152%2F7PCEt4on_400x400.jpg&w=320&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1561629357692465152%2F7PCEt4on_400x400.jpg&w=320&q=75 1x"
        style={
          {
            "color": "transparent",
            "height": "auto",
            "maxWidth": "100%",
          }
        }
        width={64}
      />
      <div
        className="text-base ml-3"
      >
        <span>
          test
        </span>
        <br />
        <span>
          test-position
        </span>
      </div>
    </div>
  </div>
</div>
`;
